File: cpp/matrix_inverse.cpp
#include <iostream>
using namespace std;

void reverseMatrix(int matrix[][3]){
	
    int matrixRows=(sizeof(matrix)/sizeof(matrix[0]));
    int matrixCols=(sizeof(matrix)/(matrixRows*sizeof(matrix[0][0])));
    cout<<"rows  cols \n",matrixRows,matrixCols ;
	int matrixRes[matrixCols][matrixRows];

	for(int i = 0; i < matrixCols; i++)
	{
	  for(int j =0; j< matrixRows; j++)
	  {
		 matrixRes[i][j]=matrix[j][i];
		 cout<< matrixRes[i][j];
	  }
	  cout<<("\n");
	}



} 

int main()
{
	int matrix[2][3]={
		{3, 2, 1},
		{4, 5, 6}
	};

	reverseMatrix(matrix);
	return 0;
}

File: python/matrix_inverse.py
matrix=[
 [3, 2, 1],
 [4, 5, 6]
]

matrix_rows=(len(matrix))
matrix_cols=(len(matrix[0]))
matrix_res=[]
for j in range(matrix_cols):
 	rows_list=[]
 	for i in range(matrix_rows):
 		rows_list.append(0)
 	matrix_res.append(rows_list)


for i in range(matrix_cols):
	for j in range(matrix_rows):
		matrix_res[i][j]=matrix[j][i]
		print(matrix_res[i][j], end="  ")
	print("\n")	